### PR TITLE
Added rules validation for email

### PR DESCRIPTION
**Pull request by steverenardbroadsign**

### CODE REVIEW

File: src/components/BadTextInput.vue [hunk 1]
```diff
@@ -15,7 +16,15 @@
 export default {
   name: "BadTextInput",
   data() {
-    return {};
+    let domainName = process.env.VUE_APP_DOMAIN_NAME;
+    return {
+      rules: {
+        isEmpty: value => !!value || value !== "" || 'Email cannot be empty',
+        hasWhitespaces: value => value.indexOf(' ') <= 0 || 'Email cannot contains spaces',
```
Comment (tibbiustin) [inline src/components/BadTextInput.vue]: Email cannot contain spaces

File: src/components/BadTextInput.vue [hunk 2]
```diff
@@ -15,7 +16,15 @@
...
+        isCorporateDomain: value => (value.endsWith(`@${domainName}`)) || (`Email should ends with @${domainName}`),
```
Comment (tibbiustin) [inline src/components/BadTextInput.vue]: Email should end with

File: tests/unit/components/BadTextInput.spec.js [hunk 3]
```diff
@@ -46,4 +48,38 @@ describe("Component BadTextInput.vue", () => {
...
+  });
+
+  it("should validation rules fails when email is empty", () => {
+    let invalidRule = underTest.vm.rules.isEmpty('');
+    expect(invalidRule).toBe("Email cannot be empty")
+  });
+
+  it("should validation rules fails when email has a whitespace", () => {
+    let invalidRule = underTest.vm.rules.hasWhitespaces('test @ domain.com');
+    expect(invalidRule).toBe("Email cannot contains spaces")
```
Comment (tibbiustin) [inline tests/unit/components/BadTextInput.spec.js]: Email cannot contain spaces

File: tests/unit/components/BadTextInput.spec.js [hunk 4]
```diff
@@ -46,4 +48,38 @@ describe("Component BadTextInput.vue", () => {
...
+  });
+
+  it("should validation rules fails when email domain is not corporate", () => {
+    let invalidRule = underTest.vm.rules.isCorporateDomain('test@domain.com');
+    expect(invalidRule).toBe(`Email should ends with @${domainName}`);
```
Comment (tibbiustin) [inline tests/unit/components/BadTextInput.spec.js]: Email should end with

File: tests/unit/components/BadTextInput.spec.js [hunk 5]
```diff
@@ -46,4 +48,38 @@ describe("Component BadTextInput.vue", () => {
     const textField = underTest.findComponent({ name: "v-text-field" });
     expect(textField.props().placeholder).toEqual("Type something");
   });
-});
+
+  it("should rules validation be successful with a correct corporate email", () => {
```
Comment (tibbiustin) [inline tests/unit/components/BadTextInput.spec.js]: what about "validation should be successful with a correct corporate email"

File: tests/unit/components/BadTextInput.spec.js [hunk 6]
```diff
@@ -46,4 +48,38 @@ describe("Component BadTextInput.vue", () => {
...
+    let hasEmailFormat = underTest.vm.rules.hasEmailFormat(validEmail);
+    let isCorporateEmail = underTest.vm.rules.isCorporateDomain(validEmail);
+    
+    expect(isRequired).toBe(true)
+    expect(hasWhitespaces).toBe(true)
+    expect(hasEmailFormat).toBe(true)
+    expect(isCorporateEmail).toBe(true)
+  });
+
+  it("should validation rules fails when email is empty", () => {
```
Comment (tibbiustin) [inline tests/unit/components/BadTextInput.spec.js]: "validation should fail when email is empty"

File: tests/unit/components/BadTextInput.spec.js [hunk 7]
```diff
@@ -46,4 +48,38 @@ describe("Component BadTextInput.vue", () => {
...
+    let invalidRule = underTest.vm.rules.isEmpty('');
+    expect(invalidRule).toBe("Email cannot be empty")
+  });
+
+  it("should validation rules fails when email has a whitespace", () => {
```
Comment (tibbiustin) [inline tests/unit/components/BadTextInput.spec.js]: validation should fail when email has a whitespace

File: tests/unit/components/BadTextInput.spec.js [hunk 8]
```diff
@@ -46,4 +48,38 @@ describe("Component BadTextInput.vue", () => {
...
+    let invalidRule = underTest.vm.rules.hasWhitespaces('test @ domain.com');
+    expect(invalidRule).toBe("Email cannot contains spaces")
+  });
+
+  it("should validation rules fails when email domain is not corporate", () => {
```
Comment (tibbiustin) [inline tests/unit/components/BadTextInput.spec.js]: validation should fail when the email's domain is not corporate

File: tests/unit/components/BadTextInput.spec.js [hunk 9]
```diff
@@ -46,4 +48,38 @@ describe("Component BadTextInput.vue", () => {
...
+    let invalidRule = underTest.vm.rules.isCorporateDomain('test@domain.com');
+    expect(invalidRule).toBe(`Email should ends with @${domainName}`);
+  });
+
+  it("should validation rules fails when email has no correct email format", () => {
```
Comment (tibbiustin) [inline tests/unit/components/BadTextInput.spec.js]: validation should fail when the email doesn't have the correct format

**Review by tibbiustin — APPROVED** [review]

File: src/components/BadTextInput.vue [hunk 10]
```diff
@@ -15,7 +16,15 @@
 export default {
   name: "BadTextInput",
   data() {
-    return {};
+    let domainName = process.env.VUE_APP_DOMAIN_NAME;
+    return {
+      rules: {
+        isEmpty: value => !!value || value !== "" || 'Email cannot be empty',
+        hasWhitespaces: value => value.indexOf(' ') <= 0 || 'Email cannot contain spaces',
```
Comment (AlexGagne) [inline src/components/BadTextInput.vue]: Trim Whitespace?

File: src/components/BadTextInput.vue [hunk 11]
```diff
@@ -15,7 +16,15 @@
 export default {
   name: "BadTextInput",
   data() {
-    return {};
+    let domainName = process.env.VUE_APP_DOMAIN_NAME;
+    return {
+      rules: {
+        isEmpty: value => !!value || value !== "" || 'Email cannot be empty',
```
Comment (AlexGagne) [inline src/components/BadTextInput.vue]: Rename to:
IsNotEmpty
DoesNotHaveWhitespace

**Review by AlexGagne — APPROVED** [review]

**Pull request merged**